hivemq4/base-image/Dockerfile

```diff
@@ -37,27 +36,35 @@ RUN set -x \
         && { command -v gpgconf && gpgconf --kill all || :; } \
         && chmod +x /usr/local/bin/gosu \
         && gosu nobody true \
-        && apt-get purge -y gpg dirmngr && rm -rf /var/lib/apt/lists/* \
-        && mkdir -p /docker-entrypoint.d
+        && apt-get purge -y gpg dirmngr && rm -rf /var/lib/apt/lists/*
+
+
+#Additional build image to unpack the zip file and change the permissions without retaining large layers just for those operations
+FROM javabase as unpack
+
+ARG HIVEMQ_VERSION=4.0.0
 
 COPY config.xml /opt/config.xml
+COPY hivemq-${HIVEMQ_VERSION}.zip /opt/hivemq-${HIVEMQ_VERSION}.zip
+RUN  unzip /opt/hivemq-${HIVEMQ_VERSION}.zip  -d /opt/ \
+     && rm -rf /opt/hivemq-${HIVEMQ_VERSION}/tools/hivemq-swarm \
+     && mv /opt/config.xml /opt/hivemq-${HIVEMQ_VERSION}/conf/config.xml \
+     && chgrp -R 0 /opt \
+     && chmod -R 770 /opt
+
+FROM javabase
+
+ARG HIVEMQ_VERSION=4.0.0
+
+COPY --from=unpack /opt/hivemq-${HIVEMQ_VERSION} /opt/hivemq
```

Comment:
So we do not use the version in the folder name anymore and drop the symlink?
I also like this more, but could this break "backwards-compatibility"?

Reply:
changed it with the simplification of the build steps.